evennia/contrib/base_systems/character_creator/example_menu.py

```diff
@@ -0,0 +1,444 @@
+    }
+]
+
+# this method will be run to create the starting objects
+def create_objects(character):
+    """do the actual object spawning"""
+    # since our example chargen saves the starting prototype to an attribute, we retrieve that here
+    proto = dict(character.db.starter_weapon)
+    # set the location to our character, so they actually have it
+    proto["location"] = character
+    # create the object
+    spawn(proto)
+    
+
+def menunode_choose_objects(caller, raw_string, **kwargs):
+    """Selecting objects to start with"""
+    char = caller.new_char
+
+    # another decision, so save the resume point
+    char.db.chargen_step = "menunode_choose_objects"
+    
+    text = dedent("""\
+        |wStarting Objects|n
+        
+        Whether it's a cosmetic outfit, a starting weapon, or a professional
+        tool kit, you probably want to let your players have a choice in
+        what objects they start out with.
+        """)
+
+    help = "An overview of what the choice affects - whether it's purely aesthetic or mechanical, and whether you can change it later - are good here."
+    
+    options = []
+    
+    for proto in _EXAMPLE_PROTOTYPES:
+        # use the key as the option description, but pass the whole prototype
+        options.append({"desc": f"Choose {_INFLECT.an(proto['key'])}", "goto": ( _set_object_choice, {"proto": proto})})
+
+    options.append({"key": ("(Back)", "back", "b"), "desc": "Go back to the previous step", "goto": "menunode_multi_choice"})
+
+    return (text, help), options
+
+def _set_object_choice(caller, raw_string, proto, **kwargs):
+    """Save the selected starting object(s)"""
+    
+    # we DON'T want to actually create the object, yet! that way players can still go back and change their mind
+    # instead, we save what object was chosen - in this case, by saving the prototype dict to the character
+    caller.new_char.db.starter_weapon = proto
+    
+    # continue to the next step
+    return "menunode_choose_name"
+
+
+#########################################################
+#                Choosing a Name
+#########################################################
+
+def menunode_choose_name(caller, raw_string, **kwargs):
+    """Name selection"""
+    char = caller.new_char
+
+    # another decision, so save the resume point
+    char.db.chargen_step = "menunode_choose_name"
+    
+    text = dedent("""\
+        |wChoosing a Name|n
+        
+        Especially for roleplaying-centric games, being able to choose your
+        character's name after deciding everything else, instead of before,
+        is really useful.
+        
+        Enter a name here to check if it's available.
+        """)
+    help = "You'll have a chance to change your mind before confirming, even if the name is free."
+    # since this is a free-text field, we just have the one
+    options = [ { "key": "_default", "goto": "menunode_check_charname" } ]
```

Comment:
Minor detail - but FYI, if you have only one option, you don't need to use a list here, but can just return the dict as-is. Keeping the list can be said to be more style-consistent though.